src/Promitor.Core.Scraping/Configuration/Serialization/ConfigurationSerializer.cs

```diff
@@ -57,7 +57,8 @@ private MetricsDeclaration InterpretYamlStream(YamlStream metricsDeclarationYaml
             switch (specVersion)
             {
                 case SpecVersion.v1:
-                    var v1Config = _v1Deserializer.Deserialize(rootNode);
+                    var errorReporter = new ErrorReporter();
```

Comment:
Nit - Is this specific for this spec or not? Otherwise we should initialize it before the switch maybe?

Reply:
I can move it - I'll have just put it there because we don't have any other versions yet, and I was initialising it as close to the object that uses it as possible.

Reply:
If you don't mind, I'll leave this for a future PR. I think in the end I changed it so the error reporter gets passed into the method instead of created here so that I could hook into Promitor's validation code and exit if validation fails.